liquid-rust-typeck/src/ty.rs

```diff
@@ -87,6 +87,8 @@ pub enum TyKind {
     Ref(RefKind, Ty),
     Param(ParamTy),
     Never,
+    /// Used internally to represent result of `discriminant` RVal
+    Discr,
```

Comment:
I think in rustc a discriminant has an integer type and you can get the discriminant for a type with https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/struct.Ty.html#method.discriminant_ty, but having a separate `Discr` type sounds like a better design to me, so let's leave it. worst case won't be too hard to remove this.